<a href="https://colab.research.google.com/github/ramonparaiba/data-science-studies/blob/main/DataSUS_LEARN.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Importação das bases de dados
Inicialmente, busquei no Site SINAN pelos dados de acidentes por animais peçonhentos.

https://portalsinan.saude.gov.br/o-sinan

Este mesmo fluxo poderá ser adotando para análises futuras

# Transformar a Base em CSV

No site https://datasus.saude.gov.br/transferencia-de-arquivos/

Deve-se buscar pelo app windows TABWIN/TABNET, a partir dele é possível transformar o arquivo .DBC em .DBF e depois em .CSV

# Importação de Bibliotecas

In [1]:
#Pandas

import pandas as pd

In [2]:
#Visualizar todas as tabelas
pd.options.display.max_columns = None

In [4]:
#Ler csv
dados = pd.read_csv("/content/ANIMBR17.csv", encoding='latin1')
dados

<ipython-input-4-963f0b00406f>:2: DtypeWarning: Columns (10,21,22) have mixed types. Specify dtype option on import or set low_memory=False.
  dados = pd.read_csv("/content/ANIMBR17.csv", encoding='latin1')


TP_NOT ID_AGRAVO  DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  \
0            2       X29    20170101   201701    2017         29      291080   
1            2       X29    20170101   201701    2017         35      352620   
2            2       X29    20170101   201701    2017         13      130360   
3            2       X29    20170101   201701    2017         41      412010   
4            2       X29    20170101   201701    2017         26      261160   
...        ...       ...         ...      ...     ...        ...         ...   
222853       2       X29    20171231   201801    2017         22      221050   
222854       2       X29    20171231   201801    2017         22      221050   
222855       2       X29    20171231   201801    2017         31      311680   
222856       2       X29    20171231   201801    2017         41      410180   
222857       2       X29    20171231   201801    2017         29      293330   

        ID_REGIONA  DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO  \
0           1381.0    20170101   201701   2012.0        4004       M   
1           1335.0    20170101   201701   1980.0        4036       F   
2           5584.0    20160515   201620   1990.0        4025       F   
3           1357.0    20170101   201701   1954.0        4062       F   
4           1497.0    20170101   201701   2013.0        4003       F   
...            ...         ...      ...      ...         ...     ...   
222853         NaN    20171231   201801   2006.0        4011       M   
222854         NaN    20171231   201801   1982.0        4035       F   
222855      1450.0    20171231   201801   1962.0        4055       M   
222856      1356.0    20171230   201752   1972.0        4045       F   
222857      1398.0    20171231   201801   1998.0        4019       F   

        CS_GESTANT  CS_RACA  CS_ESCOL_N  SG_UF  ID_MN_RESI  ID_RG_RESI  \
0              6.0      2.0        10.0   29.0    291080.0         NaN   
1              9.0      9.0         9.0   35.0    352620.0         NaN   
2              5.0      5.0         0.0   13.0    130360.0         NaN   
3              5.0      1.0         6.0   41.0    411770.0         NaN   
4              6.0      9.0        10.0   26.0    260840.0         NaN   
...            ...      ...         ...    ...         ...         ...   
222853         6.0      4.0         3.0   22.0    220060.0         NaN   
222854         5.0      4.0         3.0   22.0    221050.0         NaN   
222855         6.0      4.0         1.0   31.0    310620.0         NaN   
222856         5.0      1.0         9.0   41.0    410180.0         NaN   
222857         9.0      1.0         NaN   29.0    290120.0         NaN   

        ID_PAIS  DT_INVEST ID_OCUPA_N ANT_DT_ACI  ANT_UF  ANT_MUNIC_  \
0           1.0   20170101        NaN   20170101    29.0    291080.0   
1           1.0   20170101        NaN   20170101    35.0    352620.0   
2           1.0   20170101   612005.0   20160115    13.0    130360.0   
3           1.0   20170101   999993.0   20170101    41.0    412010.0   
4           1.0   20170101        NaN   20170101    26.0    260840.0   
...         ...        ...        ...        ...     ...         ...   
222853      1.0   20171231   999991.0   20171231    22.0    220060.0   
222854      1.0   20171231   999992.0   20171231    22.0    221050.0   
222855      1.0   20171231   782305.0   20171231    31.0    311680.0   
222856      1.0   20171231        NaN   20171230    41.0    410180.0   
222857      1.0   20171231        NaN   20171231    29.0    290120.0   

        ANT_LOCALI  ANT_TEMPO_  ANT_LOCA_1  MCLI_LOCAL  CLI_DOR  CLI_EDEMA  \
0              NaN         9.0         6.0         1.0      9.0        1.0   
1              NaN         1.0         9.0         2.0      NaN        NaN   
2              NaN         5.0         9.0         1.0      1.0        1.0   
3              NaN         1.0         9.0         1.0      1.0        2.0   
4              NaN         1.0        

#Ajustando a tabela
* As tabelas, geralmente, são geradas a partir de formulários e instrumentos próprios.
* Devemos, ajustar a tabela para ela se tornar legível
* neste caso, recorremos ao site https://portalsinan.saude.gov.br/acidente-por-animais-peconhentos
*O qual apresenta o dicionário de dados para decifrarmos a tabela.

In [5]:
#Listar as colunas
dados.columns

Index(['TP_NOT', 'ID_AGRAVO', 'DT_NOTIFIC', 'SEM_NOT', 'NU_ANO', 'SG_UF_NOT',
       'ID_MUNICIP', 'ID_REGIONA', 'DT_SIN_PRI', 'SEM_PRI', 'ANO_NASC',
       'NU_IDADE_N', 'CS_SEXO', 'CS_GESTANT', 'CS_RACA', 'CS_ESCOL_N', 'SG_UF',
       'ID_MN_RESI', 'ID_RG_RESI', 'ID_PAIS', 'DT_INVEST', 'ID_OCUPA_N',
       'ANT_DT_ACI', 'ANT_UF', 'ANT_MUNIC_', 'ANT_LOCALI', 'ANT_TEMPO_',
       'ANT_LOCA_1', 'MCLI_LOCAL', 'CLI_DOR', 'CLI_EDEMA', 'CLI_EQUIMO',
       'CLI_NECROS', 'CLI_LOCAL_', 'CLI_LOCA_1', 'MCLI_SIST', 'CLI_NEURO',
       'CLI_HEMORR', 'CLI_VAGAIS', 'CLI_MIOLIT', 'CLI_RENAL', 'CLI_OUTR_2',
       'CLI_OUTR_3', 'CLI_TEMPO_', 'TP_ACIDENT', 'ANI_TIPO_1', 'ANI_SERPEN',
       'ANI_ARANHA', 'ANI_LAGART', 'TRA_CLASSI', 'CON_SOROTE', 'NU_AMPOLAS',
       'NU_AMPOL_1', 'NU_AMPOL_8', 'NU_AMPOL_6', 'NU_AMPOL_4', 'NU_AMPO_7',
       'NU_AMPO_5', 'NU_AMPOL_9', 'NU_AMPOL_3', 'COM_LOC', 'COM_SECUND',
       'COM_NECROS', 'COM_COMPOR', 'COM_DEFICT', 'COM_APUTAC', 'COM_SISTEM',
       'COM_RENAL', 

Escolher quais colunas manter, copianda a lista gerada na célula anterior, e apagand o nome das que não nos serão úteis.

In [6]:
#Neste caso apenas removi 'TP_NOT', 'ID_AGRAVO'. este passo pode variar entre cada caso

colunas_manter = ['DT_NOTIFIC', 'SEM_NOT', 'NU_ANO', 'SG_UF_NOT',
       'ID_MUNICIP', 'ID_REGIONA', 'DT_SIN_PRI', 'SEM_PRI', 'ANO_NASC',
       'NU_IDADE_N', 'CS_SEXO', 'CS_GESTANT', 'CS_RACA', 'CS_ESCOL_N', 'SG_UF',
       'ID_MN_RESI', 'ID_RG_RESI', 'ID_PAIS', 'DT_INVEST', 'ID_OCUPA_N',
       'ANT_DT_ACI', 'ANT_UF', 'ANT_MUNIC_', 'ANT_LOCALI', 'ANT_TEMPO_',
       'ANT_LOCA_1', 'MCLI_LOCAL', 'CLI_DOR', 'CLI_EDEMA', 'CLI_EQUIMO',
       'CLI_NECROS', 'CLI_LOCAL_', 'CLI_LOCA_1', 'MCLI_SIST', 'CLI_NEURO',
       'CLI_HEMORR', 'CLI_VAGAIS', 'CLI_MIOLIT', 'CLI_RENAL', 'CLI_OUTR_2',
       'CLI_OUTR_3', 'CLI_TEMPO_', 'TP_ACIDENT', 'ANI_TIPO_1', 'ANI_SERPEN',
       'ANI_ARANHA', 'ANI_LAGART', 'TRA_CLASSI', 'CON_SOROTE', 'NU_AMPOLAS',
       'NU_AMPOL_1', 'NU_AMPOL_8', 'NU_AMPOL_6', 'NU_AMPOL_4', 'NU_AMPO_7',
       'NU_AMPO_5', 'NU_AMPOL_9', 'NU_AMPOL_3', 'COM_LOC', 'COM_SECUND',
       'COM_NECROS', 'COM_COMPOR', 'COM_DEFICT', 'COM_APUTAC', 'COM_SISTEM',
       'COM_RENAL', 'COM_EDEMA', 'COM_SEPTIC', 'COM_CHOQUE', 'DOENCA_TRA',
       'EVOLUCAO', 'DT_OBITO', 'DT_ENCERRA', 'DT_DIGITA']

In [7]:
#Substituir o dataframe atual, pelo novo apenas com as colunas que queremos manter
dados = dados[colunas_manter].reset_index(drop=True)
dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0         20170101   201701    2017         29      291080      1381.0   
1         20170101   201701    2017         35      352620      1335.0   
2         20170101   201701    2017         13      130360      5584.0   
3         20170101   201701    2017         41      412010      1357.0   
4         20170101   201701    2017         26      261160      1497.0   
...            ...      ...     ...        ...         ...         ...   
222853    20171231   201801    2017         22      221050         NaN   
222854    20171231   201801    2017         22      221050         NaN   
222855    20171231   201801    2017         31      311680      1450.0   
222856    20171231   201801    2017         41      410180      1356.0   
222857    20171231   201801    2017         29      293330      1398.0   

        DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO  CS_GESTANT  CS_RACA  \
0         20170101   201701   2012.0        4004       M         6.0      2.0   
1         20170101   201701   1980.0        4036       F         9.0      9.0   
2         20160515   201620   1990.0        4025       F         5.0      5.0   
3         20170101   201701   1954.0        4062       F         5.0      1.0   
4         20170101   201701   2013.0        4003       F         6.0      9.0   
...            ...      ...      ...         ...     ...         ...      ...   
222853    20171231   201801   2006.0        4011       M         6.0      4.0   
222854    20171231   201801   1982.0        4035       F         5.0      4.0   
222855    20171231   201801   1962.0        4055       M         6.0      4.0   
222856    20171230   201752   1972.0        4045       F         5.0      1.0   
222857    20171231   201801   1998.0        4019       F         9.0      1.0   

        CS_ESCOL_N  SG_UF  ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST  \
0             10.0   29.0    291080.0         NaN      1.0   20170101   
1              9.0   35.0    352620.0         NaN      1.0   20170101   
2              0.0   13.0    130360.0         NaN      1.0   20170101   
3              6.0   41.0    411770.0         NaN      1.0   20170101   
4             10.0   26.0    260840.0         NaN      1.0   20170101   
...            ...    ...         ...         ...      ...        ...   
222853         3.0   22.0    220060.0         NaN      1.0   20171231   
222854         3.0   22.0    221050.0         NaN      1.0   20171231   
222855         1.0   31.0    310620.0         NaN      1.0   20171231   
222856         9.0   41.0    410180.0         NaN      1.0   20171231   
222857         NaN   29.0    290120.0         NaN      1.0   20171231   

       ID_OCUPA_N ANT_DT_ACI  ANT_UF  ANT_MUNIC_  ANT_LOCALI  ANT_TEMPO_  \
0             NaN   20170101    29.0    291080.0         NaN         9.0   
1             NaN   20170101    35.0    352620.0         NaN         1.0   
2        612005.0   20160115    13.0    130360.0         NaN         5.0   
3        999993.0   20170101    41.0    412010.0         NaN         1.0   
4             NaN   20170101    26.0    260840.0         NaN         1.0   
...           ...        ...     ...         ...         ...         ...   
222853   999991.0   20171231    22.0    220060.0         NaN         1.0   
222854   999992.0   20171231    22.0    221050.0         NaN         1.0   
222855   782305.0   20171231    31.0    311680.0         NaN         2.0   
222856        NaN   20171230    41.0    410180.0         NaN         5.0   
222857        NaN   20171231    29.0    290120.0         NaN         NaN   

        ANT_LOCA_1  MCLI_LOCAL  CLI_DOR  CLI_EDEMA  CLI_EQUIMO  CLI_NECROS  \
0              6.0         1.0      9.0        1.0         9.0         9.0   
1              9.0         2.0      NaN        NaN         NaN         NaN   
2              9.0         1.0      1.0        1.0         1.0         1.0   
3              9.0         1.0      1.0     

Substituir os nomes das colunas utilizando o método **Replace**

Ao avaliar o dicionário de dados, percebe-se que muitas variáveis apresentam apenas SIM e o NÃO. para essas criaremos um mini dicionário.

In [8]:
dados['MCLI_LOCAL'] = dados['MCLI_LOCAL'].replace({
    1:'Sim',
    2:'Não',
    9:'Ignorado'
})

dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0         20170101   201701    2017         29      291080      1381.0   
1         20170101   201701    2017         35      352620      1335.0   
2         20170101   201701    2017         13      130360      5584.0   
3         20170101   201701    2017         41      412010      1357.0   
4         20170101   201701    2017         26      261160      1497.0   
...            ...      ...     ...        ...         ...         ...   
222853    20171231   201801    2017         22      221050         NaN   
222854    20171231   201801    2017         22      221050         NaN   
222855    20171231   201801    2017         31      311680      1450.0   
222856    20171231   201801    2017         41      410180      1356.0   
222857    20171231   201801    2017         29      293330      1398.0   

        DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO  CS_GESTANT  CS_RACA  \
0         20170101   201701   2012.0        4004       M         6.0      2.0   
1         20170101   201701   1980.0        4036       F         9.0      9.0   
2         20160515   201620   1990.0        4025       F         5.0      5.0   
3         20170101   201701   1954.0        4062       F         5.0      1.0   
4         20170101   201701   2013.0        4003       F         6.0      9.0   
...            ...      ...      ...         ...     ...         ...      ...   
222853    20171231   201801   2006.0        4011       M         6.0      4.0   
222854    20171231   201801   1982.0        4035       F         5.0      4.0   
222855    20171231   201801   1962.0        4055       M         6.0      4.0   
222856    20171230   201752   1972.0        4045       F         5.0      1.0   
222857    20171231   201801   1998.0        4019       F         9.0      1.0   

        CS_ESCOL_N  SG_UF  ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST  \
0             10.0   29.0    291080.0         NaN      1.0   20170101   
1              9.0   35.0    352620.0         NaN      1.0   20170101   
2              0.0   13.0    130360.0         NaN      1.0   20170101   
3              6.0   41.0    411770.0         NaN      1.0   20170101   
4             10.0   26.0    260840.0         NaN      1.0   20170101   
...            ...    ...         ...         ...      ...        ...   
222853         3.0   22.0    220060.0         NaN      1.0   20171231   
222854         3.0   22.0    221050.0         NaN      1.0   20171231   
222855         1.0   31.0    310620.0         NaN      1.0   20171231   
222856         9.0   41.0    410180.0         NaN      1.0   20171231   
222857         NaN   29.0    290120.0         NaN      1.0   20171231   

       ID_OCUPA_N ANT_DT_ACI  ANT_UF  ANT_MUNIC_  ANT_LOCALI  ANT_TEMPO_  \
0             NaN   20170101    29.0    291080.0         NaN         9.0   
1             NaN   20170101    35.0    352620.0         NaN         1.0   
2        612005.0   20160115    13.0    130360.0         NaN         5.0   
3        999993.0   20170101    41.0    412010.0         NaN         1.0   
4             NaN   20170101    26.0    260840.0         NaN         1.0   
...           ...        ...     ...         ...         ...         ...   
222853   999991.0   20171231    22.0    220060.0         NaN         1.0   
222854   999992.0   20171231    22.0    221050.0         NaN         1.0   
222855   782305.0   20171231    31.0    311680.0         NaN         2.0   
222856        NaN   20171230    41.0    410180.0         NaN         5.0   
222857        NaN   20171231    29.0    290120.0         NaN         NaN   

        ANT_LOCA_1 MCLI_LOCAL  CLI_DOR  CLI_EDEMA  CLI_EQUIMO  CLI_NECROS  \
0              6.0        Sim      9.0        1.0         9.0         9.0   
1              9.0        Não      NaN        NaN         NaN         NaN   
2              9.0        Sim      1.0        1.0         1.0         1.0   
3              9.0        Sim      1.0        2.

Buscar agora todas as séries/colunas que possuem a codificação SIM/NÃO, e vamos aplicar um laço FOR.

In [9]:
colunas_sim_nao = ['MCLI_LOCAL', 'CLI_DOR', 'CLI_EDEMA', 'CLI_EQUIMO', 'CLI_NECROS', 'CLI_LOCAL_',
                   'MCLI_SIST', 'CLI_NEURO', 'CLI_HEMORR', 'CLI_VAGAIS', 'CLI_MIOLIT', 'CLI_RENAL',
                   'CLI_OUTR_2', 'CON_SOROTE', 'COM_LOC', 'COM_SECUND', 'COM_NECROS', 'COM_COMPOR',
                   'COM_DEFICT', 'COM_APUTAC', 'COM_SISTEM', 'COM_RENAL', 'COM_EDEMA', 'COM_SEPTIC',
                   'COM_CHOQUE', 'DOENCA_TRA']

for serie in colunas_sim_nao:
  dados[serie] = dados[serie].replace({1:'Sim', 2:'Não',9:'Ignorado'})

dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0         20170101   201701    2017         29      291080      1381.0   
1         20170101   201701    2017         35      352620      1335.0   
2         20170101   201701    2017         13      130360      5584.0   
3         20170101   201701    2017         41      412010      1357.0   
4         20170101   201701    2017         26      261160      1497.0   
...            ...      ...     ...        ...         ...         ...   
222853    20171231   201801    2017         22      221050         NaN   
222854    20171231   201801    2017         22      221050         NaN   
222855    20171231   201801    2017         31      311680      1450.0   
222856    20171231   201801    2017         41      410180      1356.0   
222857    20171231   201801    2017         29      293330      1398.0   

        DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO  CS_GESTANT  CS_RACA  \
0         20170101   201701   2012.0        4004       M         6.0      2.0   
1         20170101   201701   1980.0        4036       F         9.0      9.0   
2         20160515   201620   1990.0        4025       F         5.0      5.0   
3         20170101   201701   1954.0        4062       F         5.0      1.0   
4         20170101   201701   2013.0        4003       F         6.0      9.0   
...            ...      ...      ...         ...     ...         ...      ...   
222853    20171231   201801   2006.0        4011       M         6.0      4.0   
222854    20171231   201801   1982.0        4035       F         5.0      4.0   
222855    20171231   201801   1962.0        4055       M         6.0      4.0   
222856    20171230   201752   1972.0        4045       F         5.0      1.0   
222857    20171231   201801   1998.0        4019       F         9.0      1.0   

        CS_ESCOL_N  SG_UF  ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST  \
0             10.0   29.0    291080.0         NaN      1.0   20170101   
1              9.0   35.0    352620.0         NaN      1.0   20170101   
2              0.0   13.0    130360.0         NaN      1.0   20170101   
3              6.0   41.0    411770.0         NaN      1.0   20170101   
4             10.0   26.0    260840.0         NaN      1.0   20170101   
...            ...    ...         ...         ...      ...        ...   
222853         3.0   22.0    220060.0         NaN      1.0   20171231   
222854         3.0   22.0    221050.0         NaN      1.0   20171231   
222855         1.0   31.0    310620.0         NaN      1.0   20171231   
222856         9.0   41.0    410180.0         NaN      1.0   20171231   
222857         NaN   29.0    290120.0         NaN      1.0   20171231   

       ID_OCUPA_N ANT_DT_ACI  ANT_UF  ANT_MUNIC_  ANT_LOCALI  ANT_TEMPO_  \
0             NaN   20170101    29.0    291080.0         NaN         9.0   
1             NaN   20170101    35.0    352620.0         NaN         1.0   
2        612005.0   20160115    13.0    130360.0         NaN         5.0   
3        999993.0   20170101    41.0    412010.0         NaN         1.0   
4             NaN   20170101    26.0    260840.0         NaN         1.0   
...           ...        ...     ...         ...         ...         ...   
222853   999991.0   20171231    22.0    220060.0         NaN         1.0   
222854   999992.0   20171231    22.0    221050.0         NaN         1.0   
222855   782305.0   20171231    31.0    311680.0         NaN         2.0   
222856        NaN   20171230    41.0    410180.0         NaN         5.0   
222857        NaN   20171231    29.0    290120.0         NaN         NaN   

        ANT_LOCA_1 MCLI_LOCAL   CLI_DOR CLI_EDEMA CLI_EQUIMO CLI_NECROS  \
0              6.0        Sim  Ignorado       Sim   Ignorado   Ignorado   
1              9.0        Não       NaN       NaN        NaN        NaN   
2              9.0        Sim       Sim       Sim        Sim        Sim   
3              9.0        Sim       Sim       Não       

Ajustar os formatos de Data

In [11]:
colunas_data = ['DT_NOTIFIC','DT_SIN_PRI', 'DT_INVEST', 'ANT_DT_ACI', 'DT_OBITO', 'DT_ENCERRA', 'DT_DIGITA' ]

for serie in colunas_data:
  dados[serie] = pd.to_datetime(dados[serie], format='%Y%m%d', errors='coerce')

dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0      2017-01-01   201701    2017         29      291080      1381.0   
1      2017-01-01   201701    2017         35      352620      1335.0   
2      2017-01-01   201701    2017         13      130360      5584.0   
3      2017-01-01   201701    2017         41      412010      1357.0   
4      2017-01-01   201701    2017         26      261160      1497.0   
...           ...      ...     ...        ...         ...         ...   
222853 2017-12-31   201801    2017         22      221050         NaN   
222854 2017-12-31   201801    2017         22      221050         NaN   
222855 2017-12-31   201801    2017         31      311680      1450.0   
222856 2017-12-31   201801    2017         41      410180      1356.0   
222857 2017-12-31   201801    2017         29      293330      1398.0   

       DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO  CS_GESTANT  CS_RACA  \
0      2017-01-01   201701   2012.0        4004       M         6.0      2.0   
1      2017-01-01   201701   1980.0        4036       F         9.0      9.0   
2      2016-05-15   201620   1990.0        4025       F         5.0      5.0   
3      2017-01-01   201701   1954.0        4062       F         5.0      1.0   
4      2017-01-01   201701   2013.0        4003       F         6.0      9.0   
...           ...      ...      ...         ...     ...         ...      ...   
222853 2017-12-31   201801   2006.0        4011       M         6.0      4.0   
222854 2017-12-31   201801   1982.0        4035       F         5.0      4.0   
222855 2017-12-31   201801   1962.0        4055       M         6.0      4.0   
222856 2017-12-30   201752   1972.0        4045       F         5.0      1.0   
222857 2017-12-31   201801   1998.0        4019       F         9.0      1.0   

        CS_ESCOL_N  SG_UF  ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST  \
0             10.0   29.0    291080.0         NaN      1.0 2017-01-01   
1              9.0   35.0    352620.0         NaN      1.0 2017-01-01   
2              0.0   13.0    130360.0         NaN      1.0 2017-01-01   
3              6.0   41.0    411770.0         NaN      1.0 2017-01-01   
4             10.0   26.0    260840.0         NaN      1.0 2017-01-01   
...            ...    ...         ...         ...      ...        ...   
222853         3.0   22.0    220060.0         NaN      1.0 2017-12-31   
222854         3.0   22.0    221050.0         NaN      1.0 2017-12-31   
222855         1.0   31.0    310620.0         NaN      1.0 2017-12-31   
222856         9.0   41.0    410180.0         NaN      1.0 2017-12-31   
222857         NaN   29.0    290120.0         NaN      1.0 2017-12-31   

       ID_OCUPA_N ANT_DT_ACI  ANT_UF  ANT_MUNIC_  ANT_LOCALI  ANT_TEMPO_  \
0             NaN 2017-01-01    29.0    291080.0         NaN         9.0   
1             NaN 2017-01-01    35.0    352620.0         NaN         1.0   
2        612005.0 2016-01-15    13.0    130360.0         NaN         5.0   
3        999993.0 2017-01-01    41.0    412010.0         NaN         1.0   
4             NaN 2017-01-01    26.0    260840.0         NaN         1.0   
...           ...        ...     ...         ...         ...         ...   
222853   999991.0 2017-12-31    22.0    220060.0         NaN         1.0   
222854   999992.0 2017-12-31    22.0    221050.0         NaN         1.0   
222855   782305.0 2017-12-31    31.0    311680.0         NaN         2.0   
222856        NaN 2017-12-30    41.0    410180.0         NaN         5.0   
222857        NaN 2017-12-31    29.0    290120.0         NaN         NaN   

        ANT_LOCA_1 MCLI_LOCAL   CLI_DOR CLI_EDEMA CLI_EQUIMO CLI_NECROS  \
0              6.0        Sim  Ignorado       Sim   Ignorado   Ignorado   
1              9.0        Não       NaN       NaN        NaN        NaN   
2              9.0        Sim       Sim       Sim        Sim        Sim   
3              9.0        Sim       Sim       Não        Não        Não   
4   

In [10]:
dados.columns

Index(['DT_NOTIFIC', 'SEM_NOT', 'NU_ANO', 'SG_UF_NOT', 'ID_MUNICIP',
       'ID_REGIONA', 'DT_SIN_PRI', 'SEM_PRI', 'ANO_NASC', 'NU_IDADE_N',
       'CS_SEXO', 'CS_GESTANT', 'CS_RACA', 'CS_ESCOL_N', 'SG_UF', 'ID_MN_RESI',
       'ID_RG_RESI', 'ID_PAIS', 'DT_INVEST', 'ID_OCUPA_N', 'ANT_DT_ACI',
       'ANT_UF', 'ANT_MUNIC_', 'ANT_LOCALI', 'ANT_TEMPO_', 'ANT_LOCA_1',
       'MCLI_LOCAL', 'CLI_DOR', 'CLI_EDEMA', 'CLI_EQUIMO', 'CLI_NECROS',
       'CLI_LOCAL_', 'CLI_LOCA_1', 'MCLI_SIST', 'CLI_NEURO', 'CLI_HEMORR',
       'CLI_VAGAIS', 'CLI_MIOLIT', 'CLI_RENAL', 'CLI_OUTR_2', 'CLI_OUTR_3',
       'CLI_TEMPO_', 'TP_ACIDENT', 'ANI_TIPO_1', 'ANI_SERPEN', 'ANI_ARANHA',
       'ANI_LAGART', 'TRA_CLASSI', 'CON_SOROTE', 'NU_AMPOLAS', 'NU_AMPOL_1',
       'NU_AMPOL_8', 'NU_AMPOL_6', 'NU_AMPOL_4', 'NU_AMPO_7', 'NU_AMPO_5',
       'NU_AMPOL_9', 'NU_AMPOL_3', 'COM_LOC', 'COM_SECUND', 'COM_NECROS',
       'COM_COMPOR', 'COM_DEFICT', 'COM_APUTAC', 'COM_SISTEM', 'COM_RENAL',
       'COM_EDEMA', 'COM_SEPTI

### Reescrita das codificações do dicionário de dados.
Nesta etapa, deve-se reescrever TODAS as codificações do dicionário de dados. Para isso, usamos novamente a função REPLACE

In [21]:
dados['ANT_TEMPO_'] = dados['ANT_TEMPO_'].replace({
    1: '0 - 1h',
    2: '1 - 3h',
    3: '3 - 6h',
    4: '6 - 12h',
    5: '12 e 24h',
    6: '24 e +h',
    9: 'Ignorado'
})

dados['ANT_LOCA_1'] = dados['ANT_LOCA_1'].replace({
    1: 'Cabeça',
    2: 'Braço',
    3: 'Ante-Braço',
    4: 'Mão',
    5: 'Dedo da Mão',
    6: 'tronco',
    7: 'Coxa',
    8: 'Perna',
    9: 'Pé',
    10: 'Dedo do Pé',
    99: 'Ignorado'
})

dados['CLI_TEMPO_'] = dados['CLI_TEMPO_'].replace({
    1 : 'Normal',
    2 : 'Alterado',
    9 : 'Não realizado'
})

dados['TP_ACIDENT'] = dados['TP_ACIDENT'].replace({
    1 : 'Serpente',
    2 : 'Aranha',
    3 : 'Escorpião',
    4 : 'Lagarta',
    5 : 'Abelha',
    6 : 'Outros',
    9 : 'Ignorado'
})

dados['ANI_SERPEN'] = dados['ANI_SERPEN'].replace({
      1: 'Botrópico',
      2: 'Crotálico',
      3: 'Elapídico',
      4: 'Laquético',
      5: 'Serpente não peçonhenta',
      9: 'Ignorado'
      })

dados['ANI_ARANHA'] = dados['ANI_ARANHA'].replace({
      1: 'Foneutrismo',
      2: 'Loxoscelismo',
      3: 'Latrodectismo',
      4: 'Outra aranha',
      9: 'Ignorado'
      })

dados['ANI_LAGART'] = dados['ANI_LAGART'].replace({
      1: 'Lonomia',
      2: 'Outra lagarta',
      9: 'Ignorado',
      })

dados['TRA_CLASSI'] = dados['TRA_CLASSI'].replace({
      1: 'Leve',
      2: 'Moderado',
      3: 'Grave',
      9: 'Ignorado'
      })

dados['EVOLUCAO'] = dados['EVOLUCAO'].replace({
      1: 'Cura',
      2: 'Óbito por acidente por animais peçonhentos',
      3: 'Óbito por outras causas',
      9: 'Ignorado'
      })

dados['CS_GESTANT'] = dados['CS_GESTANT'].replace({
      1: '1º Trimestre',
      2: '2º Trimestre',
      3: '3º Trimestre',
      4: 'Idade gestacional Ignorada',
      5: 'Não',
      6: 'Não se aplica',
      9: 'Ignorado'
      })
dados['CS_RACA'] = dados['CS_RACA'].replace({
      1: 'Branca',
      2: 'Preta',
      3: 'Amarela',
      4: 'Parda',
      5: 'Indígena',
      9: 'Ignorado',
      })
dados['CS_ESCOL_N'] = dados['CS_ESCOL_N'].replace({
      0: 'Analfabeto',
      1: '1ª a 4ª série incompleta do EF (antigo primário ou 1º grau)',
      2: '4ª série completa do EF (antigo primário ou 1º grau)',
      3: '5ª à 8ª série incompleta do EF (antigo ginásio ou 1º grau)',
      4: 'Ensino fundamental completo (antigo ginásio ou 1º grau)',
      5: 'Ensino médio incompleto (antigo colegial ou 2º grau)',
      6: 'Ensino médio completo (antigo colegial ou 2º grau)',
      7: 'Educação superior incompleta',
      8: 'Educação superior completa',
      9: 'Ignorado',
      10: 'Não se aplica'
      })



dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0      2017-01-01   201701    2017         29      291080      1381.0   
1      2017-01-01   201701    2017         35      352620      1335.0   
2      2017-01-01   201701    2017         13      130360      5584.0   
3      2017-01-01   201701    2017         41      412010      1357.0   
4      2017-01-01   201701    2017         26      261160      1497.0   
...           ...      ...     ...        ...         ...         ...   
222853 2017-12-31   201801    2017         22      221050         NaN   
222854 2017-12-31   201801    2017         22      221050         NaN   
222855 2017-12-31   201801    2017         31      311680      1450.0   
222856 2017-12-31   201801    2017         41      410180      1356.0   
222857 2017-12-31   201801    2017         29      293330      1398.0   

       DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO     CS_GESTANT  \
0      2017-01-01   201701   2012.0        4004       M  Não se aplica   
1      2017-01-01   201701   1980.0        4036       F       Ignorado   
2      2016-05-15   201620   1990.0        4025       F            Não   
3      2017-01-01   201701   1954.0        4062       F            Não   
4      2017-01-01   201701   2013.0        4003       F  Não se aplica   
...           ...      ...      ...         ...     ...            ...   
222853 2017-12-31   201801   2006.0        4011       M  Não se aplica   
222854 2017-12-31   201801   1982.0        4035       F            Não   
222855 2017-12-31   201801   1962.0        4055       M  Não se aplica   
222856 2017-12-30   201752   1972.0        4045       F            Não   
222857 2017-12-31   201801   1998.0        4019       F       Ignorado   

         CS_RACA                                         CS_ESCOL_N  SG_UF  \
0          Preta                                      Não se aplica   29.0   
1       Ignorado                                           Ignorado   35.0   
2       Indígena                                         Analfabeto   13.0   
3         Branca  Ensino médio completo (antigo colegial ou 2º g...   41.0   
4       Ignorado                                      Não se aplica   26.0   
...          ...                                                ...    ...   
222853     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222854     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222855     Parda  1ª a 4ª série incompleta do EF (antigo primári...   31.0   
222856    Branca                                           Ignorado   41.0   
222857    Branca                                                NaN   29.0   

        ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST ID_OCUPA_N ANT_DT_ACI  \
0         291080.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
1         352620.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
2         130360.0         NaN      1.0 2017-01-01   612005.0 2016-01-15   
3         411770.0         NaN      1.0 2017-01-01   999993.0 2017-01-01   
4         260840.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
...            ...         ...      ...        ...        ...        ...   
222853    220060.0         NaN      1.0 2017-12-31   999991.0 2017-12-31   
222854    221050.0         NaN      1.0 2017-12-31   999992.0 2017-12-31   
222855    310620.0         NaN      1.0 2017-12-31   782305.0 2017-12-31   
222856    410180.0         NaN      1.0 2017-12-31        NaN 2017-12-30   
222857    290120.0         NaN      1.0 2017-12-31        NaN 2017-12-31   

        ANT_UF  ANT_MUNIC_  ANT_LOCALI ANT_TEMPO_  ANT_LOCA_1 MCLI_LOCAL  \
0         29.0    291080.0         NaN   Ignorado      tronco        Sim   
1         35.0    352620.0         NaN     0 - 1h          Pé        Não   
2         13.0    130360.0         NaN   12 e 24h          Pé        Sim   
3         41.0    412010.0         NaN     0 - 1h          Pé        Sim   
4         2

## TRATAR A COLUNA NU_IDADE_N


A idade no SINAN é utilizada de forma codificada, obedecendo o seguinte critério:
1. Hora
2. Dia
3. Mês
4. Ano

Exemplos:

* 3007 - Sete meses
* 4035 - Trinta e cinco anos



In [23]:
idades = []

for idade in dados['NU_IDADE_N']:
  if idade > 4000:
    idades.append(idade - 4000)
  else:
    idades.append(0)

dados['NU_IDADE_N_'] = idades

dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0      2017-01-01   201701    2017         29      291080      1381.0   
1      2017-01-01   201701    2017         35      352620      1335.0   
2      2017-01-01   201701    2017         13      130360      5584.0   
3      2017-01-01   201701    2017         41      412010      1357.0   
4      2017-01-01   201701    2017         26      261160      1497.0   
...           ...      ...     ...        ...         ...         ...   
222853 2017-12-31   201801    2017         22      221050         NaN   
222854 2017-12-31   201801    2017         22      221050         NaN   
222855 2017-12-31   201801    2017         31      311680      1450.0   
222856 2017-12-31   201801    2017         41      410180      1356.0   
222857 2017-12-31   201801    2017         29      293330      1398.0   

       DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO     CS_GESTANT  \
0      2017-01-01   201701   2012.0        4004       M  Não se aplica   
1      2017-01-01   201701   1980.0        4036       F       Ignorado   
2      2016-05-15   201620   1990.0        4025       F            Não   
3      2017-01-01   201701   1954.0        4062       F            Não   
4      2017-01-01   201701   2013.0        4003       F  Não se aplica   
...           ...      ...      ...         ...     ...            ...   
222853 2017-12-31   201801   2006.0        4011       M  Não se aplica   
222854 2017-12-31   201801   1982.0        4035       F            Não   
222855 2017-12-31   201801   1962.0        4055       M  Não se aplica   
222856 2017-12-30   201752   1972.0        4045       F            Não   
222857 2017-12-31   201801   1998.0        4019       F       Ignorado   

         CS_RACA                                         CS_ESCOL_N  SG_UF  \
0          Preta                                      Não se aplica   29.0   
1       Ignorado                                           Ignorado   35.0   
2       Indígena                                         Analfabeto   13.0   
3         Branca  Ensino médio completo (antigo colegial ou 2º g...   41.0   
4       Ignorado                                      Não se aplica   26.0   
...          ...                                                ...    ...   
222853     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222854     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222855     Parda  1ª a 4ª série incompleta do EF (antigo primári...   31.0   
222856    Branca                                           Ignorado   41.0   
222857    Branca                                                NaN   29.0   

        ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST ID_OCUPA_N ANT_DT_ACI  \
0         291080.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
1         352620.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
2         130360.0         NaN      1.0 2017-01-01   612005.0 2016-01-15   
3         411770.0         NaN      1.0 2017-01-01   999993.0 2017-01-01   
4         260840.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
...            ...         ...      ...        ...        ...        ...   
222853    220060.0         NaN      1.0 2017-12-31   999991.0 2017-12-31   
222854    221050.0         NaN      1.0 2017-12-31   999992.0 2017-12-31   
222855    310620.0         NaN      1.0 2017-12-31   782305.0 2017-12-31   
222856    410180.0         NaN      1.0 2017-12-31        NaN 2017-12-30   
222857    290120.0         NaN      1.0 2017-12-31        NaN 2017-12-31   

        ANT_UF  ANT_MUNIC_  ANT_LOCALI ANT_TEMPO_  ANT_LOCA_1 MCLI_LOCAL  \
0         29.0    291080.0         NaN   Ignorado      tronco        Sim   
1         35.0    352620.0         NaN     0 - 1h          Pé        Não   
2         13.0    130360.0         NaN   12 e 24h          Pé        Sim   
3         41.0    412010.0         NaN     0 - 1h          Pé        Sim   
4         2

# Decodificando ID_MUNICIP

A base de dados do SINAN usa o código do IBGE para nome dos municípios

Deve ser usada uma tabela correspondente de código de UF

In [43]:
municipios = pd.read_csv('/content/municipios.csv')

municipios

COD UF      COD                   NOME
0         11  1100015  Alta Floresta D'oeste
1         11  1100023              Ariquemes
2         11  1100031                 Cabixi
3         11  1100049                 Cacoal
4         11  1100056             Cerejeiras
...      ...      ...                    ...
5565      52  5222005             Vianópolis
5566      52  5222054         Vicentinópolis
5567      52  5222203               Vila Boa
5568      52  5222302          Vila Propício
5569      53  5300108               Brasília

[5570 rows x 3 columns]

Converter em dicionário no qual codigo do municipio : "Nome do Município"

In [44]:
municipios['IBGE'] = (municipios['COD'] // 10).astype('int64')
municipios

COD UF      COD                   NOME    IBGE
0         11  1100015  Alta Floresta D'oeste  110001
1         11  1100023              Ariquemes  110002
2         11  1100031                 Cabixi  110003
3         11  1100049                 Cacoal  110004
4         11  1100056             Cerejeiras  110005
...      ...      ...                    ...     ...
5565      52  5222005             Vianópolis  522200
5566      52  5222054         Vicentinópolis  522205
5567      52  5222203               Vila Boa  522220
5568      52  5222302          Vila Propício  522230
5569      53  5300108               Brasília  530010

[5570 rows x 4 columns]

In [45]:
municipios = municipios[['IBGE', 'NOME']]
municipios = municipios.set_index('IBGE')

municipios

NOME
IBGE                         
110001  Alta Floresta D'oeste
110002              Ariquemes
110003                 Cabixi
110004                 Cacoal
110005             Cerejeiras
...                       ...
522200             Vianópolis
522205         Vicentinópolis
522220               Vila Boa
522230          Vila Propício
530010               Brasília

[5570 rows x 1 columns]

In [ ]:
#Transformar em dicionário
dicionario_municipios = municipios.to_dict()['NOME']
dicionario_municipios

In [ ]:
dados.info()

In [ ]:
#Usando a função map
dados['ID_MUNICIP_NOME'] = dados['ID_MUNICIP'].map(dicionario_municipios)
dados['ID_MN_RESI_NOME'] = dados['ID_MN_RESI'].map(dicionario_municipios)
dados['ANT_MUNIC_NOME'] = dados['ANT_MUNIC_'].map(dicionario_municipios)

dados


## Ocupação

In [79]:
#Identifiquei que a coluna ID_OCUPA_N é do tipo object, então, preciso transforma-la em inteiro
## Para tal, vou criar uma nova série ID_OCUPA_N_INT
#pd.to_numeric(df['COD'], errors='coerce').astype('Int64')
dados['ID_OCUPA_N_INT'] = pd.to_numeric(dados['ID_OCUPA_N'], errors = 'coerce').astype('Int64')
dados['ID_OCUPA_N_INT'] = dados['ID_OCUPA_N_INT'].astype(str)

In [93]:
ocupacao = pd.read_csv('https://raw.githubusercontent.com/andrejarenkow/csv/refs/heads/master/CBO2002%20-%20Ocupacao.csv')
ocupacao = ocupacao.set_index('CODIGO')
dicionario_ocupacao = ocupacao.to_dict()['TITULO']
dicionario_ocupacao

dados['ID_OCUPA_N_NOME'] = dados['ID_OCUPA_N_INT'].map(dicionario_ocupacao)
dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0      2017-01-01   201701    2017         29      291080      1381.0   
1      2017-01-01   201701    2017         35      352620      1335.0   
2      2017-01-01   201701    2017         13      130360      5584.0   
3      2017-01-01   201701    2017         41      412010      1357.0   
4      2017-01-01   201701    2017         26      261160      1497.0   
...           ...      ...     ...        ...         ...         ...   
222853 2017-12-31   201801    2017         22      221050         NaN   
222854 2017-12-31   201801    2017         22      221050         NaN   
222855 2017-12-31   201801    2017         31      311680      1450.0   
222856 2017-12-31   201801    2017         41      410180      1356.0   
222857 2017-12-31   201801    2017         29      293330      1398.0   

       DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO     CS_GESTANT  \
0      2017-01-01   201701   2012.0        4004       M  Não se aplica   
1      2017-01-01   201701   1980.0        4036       F       Ignorado   
2      2016-05-15   201620   1990.0        4025       F            Não   
3      2017-01-01   201701   1954.0        4062       F            Não   
4      2017-01-01   201701   2013.0        4003       F  Não se aplica   
...           ...      ...      ...         ...     ...            ...   
222853 2017-12-31   201801   2006.0        4011       M  Não se aplica   
222854 2017-12-31   201801   1982.0        4035       F            Não   
222855 2017-12-31   201801   1962.0        4055       M  Não se aplica   
222856 2017-12-30   201752   1972.0        4045       F            Não   
222857 2017-12-31   201801   1998.0        4019       F       Ignorado   

         CS_RACA                                         CS_ESCOL_N  SG_UF  \
0          Preta                                      Não se aplica   29.0   
1       Ignorado                                           Ignorado   35.0   
2       Indígena                                         Analfabeto   13.0   
3         Branca  Ensino médio completo (antigo colegial ou 2º g...   41.0   
4       Ignorado                                      Não se aplica   26.0   
...          ...                                                ...    ...   
222853     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222854     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222855     Parda  1ª a 4ª série incompleta do EF (antigo primári...   31.0   
222856    Branca                                           Ignorado   41.0   
222857    Branca                                                NaN   29.0   

        ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST ID_OCUPA_N ANT_DT_ACI  \
0         291080.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
1         352620.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
2         130360.0         NaN      1.0 2017-01-01   612005.0 2016-01-15   
3         411770.0         NaN      1.0 2017-01-01   999993.0 2017-01-01   
4         260840.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
...            ...         ...      ...        ...        ...        ...   
222853    220060.0         NaN      1.0 2017-12-31   999991.0 2017-12-31   
222854    221050.0         NaN      1.0 2017-12-31   999992.0 2017-12-31   
222855    310620.0         NaN      1.0 2017-12-31   782305.0 2017-12-31   
222856    410180.0         NaN      1.0 2017-12-31        NaN 2017-12-30   
222857    290120.0         NaN      1.0 2017-12-31        NaN 2017-12-31   

        ANT_UF  ANT_MUNIC_  ANT_LOCALI ANT_TEMPO_  ANT_LOCA_1 MCLI_LOCAL  \
0         29.0    291080.0         NaN   Ignorado      tronco        Sim   
1         35.0    352620.0         NaN     0 - 1h          Pé        Não   
2         13.0    130360.0         NaN   12 e 24h          Pé        Sim   
3         41.0    412010.0         NaN     0 - 1h          Pé        Sim   
4         2

In [96]:
uf = pd.read_csv('https://raw.githubusercontent.com/andrejarenkow/csv/refs/heads/master/cod_uf.csv')
uf = uf.set_index('Código UF')
dicionario_UF = uf.to_dict()['UF']


dados['SG_UF_NOME'] = dados['SG_UF'].map(dicionario_UF)
dados['SG_UF_NOT_NOME'] = dados['SG_UF_NOT'].map(dicionario_UF)
dados['ANT_UF_NOME'] = dados['ANT_UF'].map(dicionario_UF)

dados

DT_NOTIFIC  SEM_NOT  NU_ANO  SG_UF_NOT  ID_MUNICIP  ID_REGIONA  \
0      2017-01-01   201701    2017         29      291080      1381.0   
1      2017-01-01   201701    2017         35      352620      1335.0   
2      2017-01-01   201701    2017         13      130360      5584.0   
3      2017-01-01   201701    2017         41      412010      1357.0   
4      2017-01-01   201701    2017         26      261160      1497.0   
...           ...      ...     ...        ...         ...         ...   
222853 2017-12-31   201801    2017         22      221050         NaN   
222854 2017-12-31   201801    2017         22      221050         NaN   
222855 2017-12-31   201801    2017         31      311680      1450.0   
222856 2017-12-31   201801    2017         41      410180      1356.0   
222857 2017-12-31   201801    2017         29      293330      1398.0   

       DT_SIN_PRI  SEM_PRI ANO_NASC  NU_IDADE_N CS_SEXO     CS_GESTANT  \
0      2017-01-01   201701   2012.0        4004       M  Não se aplica   
1      2017-01-01   201701   1980.0        4036       F       Ignorado   
2      2016-05-15   201620   1990.0        4025       F            Não   
3      2017-01-01   201701   1954.0        4062       F            Não   
4      2017-01-01   201701   2013.0        4003       F  Não se aplica   
...           ...      ...      ...         ...     ...            ...   
222853 2017-12-31   201801   2006.0        4011       M  Não se aplica   
222854 2017-12-31   201801   1982.0        4035       F            Não   
222855 2017-12-31   201801   1962.0        4055       M  Não se aplica   
222856 2017-12-30   201752   1972.0        4045       F            Não   
222857 2017-12-31   201801   1998.0        4019       F       Ignorado   

         CS_RACA                                         CS_ESCOL_N  SG_UF  \
0          Preta                                      Não se aplica   29.0   
1       Ignorado                                           Ignorado   35.0   
2       Indígena                                         Analfabeto   13.0   
3         Branca  Ensino médio completo (antigo colegial ou 2º g...   41.0   
4       Ignorado                                      Não se aplica   26.0   
...          ...                                                ...    ...   
222853     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222854     Parda  5ª à 8ª série incompleta do EF (antigo ginásio...   22.0   
222855     Parda  1ª a 4ª série incompleta do EF (antigo primári...   31.0   
222856    Branca                                           Ignorado   41.0   
222857    Branca                                                NaN   29.0   

        ID_MN_RESI  ID_RG_RESI  ID_PAIS  DT_INVEST ID_OCUPA_N ANT_DT_ACI  \
0         291080.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
1         352620.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
2         130360.0         NaN      1.0 2017-01-01   612005.0 2016-01-15   
3         411770.0         NaN      1.0 2017-01-01   999993.0 2017-01-01   
4         260840.0         NaN      1.0 2017-01-01        NaN 2017-01-01   
...            ...         ...      ...        ...        ...        ...   
222853    220060.0         NaN      1.0 2017-12-31   999991.0 2017-12-31   
222854    221050.0         NaN      1.0 2017-12-31   999992.0 2017-12-31   
222855    310620.0         NaN      1.0 2017-12-31   782305.0 2017-12-31   
222856    410180.0         NaN      1.0 2017-12-31        NaN 2017-12-30   
222857    290120.0         NaN      1.0 2017-12-31        NaN 2017-12-31   

        ANT_UF  ANT_MUNIC_  ANT_LOCALI ANT_TEMPO_  ANT_LOCA_1 MCLI_LOCAL  \
0         29.0    291080.0         NaN   Ignorado      tronco        Sim   
1         35.0    352620.0         NaN     0 - 1h          Pé        Não   
2         13.0    130360.0         NaN   12 e 24h          Pé        Sim   
3         41.0    412010.0         NaN     0 - 1h          Pé        Sim   
4         2

In [94]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 222858 entries, 0 to 222857
Data columns (total 80 columns):
 #   Column           Non-Null Count   Dtype         
---  ------           --------------   -----         
 0   DT_NOTIFIC       222858 non-null  datetime64[ns]
 1   SEM_NOT          222858 non-null  int64         
 2   NU_ANO           222858 non-null  int64         
 3   SG_UF_NOT        222858 non-null  int64         
 4   ID_MUNICIP       222858 non-null  int64         
 5   ID_REGIONA       212929 non-null  float64       
 6   DT_SIN_PRI       222858 non-null  datetime64[ns]
 7   SEM_PRI          222858 non-null  int64         
 8   ANO_NASC         219324 non-null  object        
 9   NU_IDADE_N       222858 non-null  int64         
 10  CS_SEXO          222858 non-null  object        
 11  CS_GESTANT       222851 non-null  object        
 12  CS_RACA          219292 non-null  object        
 13  CS_ESCOL_N       199978 non-null  object        
 14  SG_UF            222